File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []
        for x,y in points:
            heapq.heappush(heap,((math.sqrt(x**2 + y**2)),[x,y]))
            
        output = []
        i = 1
        while heap and i <= k:
            h= heapq.heappop(heap)
            output.append(h[1])
            i +=1
        return output

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.ts
function kClosest(points: number[][], k: number): number[][] {
    let dist = [];
    for (let i = 0; i < points.length; i++) {
        dist.push([Math.sqrt(points[i][0] ** 2 + points[i][1] ** 2), i]);
    }
    
    dist.sort((a, b) => a[0] - b[0]);
    
    let res = [];
    for (let i = 0; i < k; i++) {
        res.push(points[dist[i][1]]);
    }
    
    return res;
};